### PR TITLE
Refactor error handling to controller advice

### PR DESCRIPTION
**Pull request by pgesek**
This is cleaner and does not require redundant
exception catching. Moreover logger formats are now static
as they always should have been. I also cleaned up some
IntelliJ warnings.

### CODE REVIEW

File: src/main/java/org/benetech/mathshare/controller/ProblemSetController.java [hunk 1]
```diff
@@ -33,85 +29,59 @@
...
-            } else {
-                logger.error("Default problem set wasn't found");
-            }
-        } catch (IllegalArgumentException e) {
-            logger.error(e.getMessage(), e);
+        String result = problemSetService.getDefaultProblemSetRevisionCode();
+        if (result != null) {
+            return new ResponseEntity<>(result, HttpStatus.OK);
+        } else {
+            logger.error("Default problem set wasn't found");
```
Comment (oskarhinc) [inline src/main/java/org/benetech/mathshare/controller/ProblemSetController.java]: Could you change this message to "default problem set revision..." by the way?

File: src/main/java/org/benetech/mathshare/controller/GlobalErrorHandling.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.WebRequest;
+
+@ControllerAdvice
+public class GlobalErrorHandling {
+
+    private final Logger logger = LoggerFactory.getLogger(GlobalErrorHandling.class);
+
+    @ExceptionHandler(IllegalArgumentException.class)
```
Comment (oskarhinc) [inline src/main/java/org/benetech/mathshare/controller/GlobalErrorHandling.java]: Have you forgotten about HttpMessageNotReadableException or you omitted it purposely?

Comment (pgesek) [reply src/main/java/org/benetech/mathshare/controller/GlobalErrorHandling.java]: who is throwing it and why should we catch it?

Comment (oskarhinc) [reply src/main/java/org/benetech/mathshare/controller/GlobalErrorHandling.java]: You're right, it's currently not thrown, so we should omit it.

**Pull request merged**